fix(build): update to latest from newrelic-forks/repolinter

jbeveland27 opened the pull request:
Attempting to get the build going again

jbeveland27 committed:
fix(build): use branch of repolinter and reinstall from npm

+ use node 16

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/newrelic/repolinter-action?pullRequest=14) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/newrelic/repolinter-action?pullRequest=14) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/newrelic/repolinter-action?pullRequest=14) it.</sub>

jbeveland27 committed:
fix: only build on ubuntu latest with node 16

jbeveland27 committed:
chore: use checkout v3

jbeveland27 committed:
fix: remove perl-base file

jbeveland27 committed:
chore: update deps

jbeveland27 committed:
fix: sync updates from upstream repolinter

jbeveland27 committed:
fix: simplify build

jbeveland27 committed:
chore: update third-party manifest and notices

jbeveland27 committed:
chore: package up the build

jbeveland27 committed:
feat: split into separate PR/release workflows

+ update NPM caching to not use set-output
+ update all setup node usage to node 16
+ fix third-party-notices gen to so checkout action fetches last 2 commits and does the comparison accurately
+ use semantic-release-action@v3 to bypass bug with v20.0.0
+ update ghaction-docker-meta action to use v2 tag

jbeveland27 committed:
test: publish pre-release version to ghcr

jbeveland27 committed:
fix: remove guards

jbeveland27 committed:
test: use different event type to test container publishing

jbeveland27 committed:
test: use push event to test container publishing

jbeveland27 commented:
Update:
* https://github.com/newrelic-forks/repolinter has been synced with the upstream repo and is building successfully
* `repolinter-action`'s build is now successful again

Next: attempting to run this branch with a test workflow to make sure behavior is consistent.

jbeveland27 committed:
test: only run release and use GITHUB_TOKEN

jbeveland27 committed:
fix: check out from current branch

jbeveland27 committed:
fix: use GITHUB_TOKEN to publish to ghcr

jbeveland27 committed:
fix: re-target the PR workflow to run on pull_request

jbeveland27 committed:
chore: comment-out the pre-release job

jbeveland27 committed:
chore: use latest version of repolinter
